src/include/rados/rados_types.h

```diff
@@ -18,6 +18,17 @@ struct obj_watch_t {
   uint32_t timeout_seconds;
 }; 
 
+struct notify_ack_t {
+  uint64_t notifier_id;
+  uint64_t cookie;
+  char *payload;
```

Comment:
Nit: same comment here. You need to be able to support non-null terminated data

Reply:
ok -- (buffer, len) would suffice then.